searchcore/src/vespa/searchcore/proton/matching/termdatafromnode.cpp

```diff
@@ -19,29 +18,30 @@ struct TermDataFromTermVisitor
         data = &n;
     }
 
-    virtual void visit(ProtonAnd &) override {}
-    virtual void visit(ProtonAndNot &) override {}
-    virtual void visit(ProtonNear &) override {}
-    virtual void visit(ProtonONear &) override {}
-    virtual void visit(ProtonOr &) override {}
-    virtual void visit(ProtonRank &) override {}
-    virtual void visit(ProtonWeakAnd &) override {}
-
-    virtual void visit(ProtonWeightedSetTerm &n) override { visitTerm(n); }
-    virtual void visit(ProtonDotProduct &n) override { visitTerm(n); }
-    virtual void visit(ProtonWandTerm &n) override { visitTerm(n); }
-    virtual void visit(ProtonPhrase &n) override { visitTerm(n); }
-    virtual void visit(ProtonEquiv &n) override { visitTerm(n); }
-
-    virtual void visit(ProtonNumberTerm &n) override { visitTerm(n); }
-    virtual void visit(ProtonLocationTerm &n) override { visitTerm(n); }
-    virtual void visit(ProtonPrefixTerm &n) override { visitTerm(n); }
-    virtual void visit(ProtonRangeTerm &n) override { visitTerm(n); }
-    virtual void visit(ProtonStringTerm &n) override { visitTerm(n); }
-    virtual void visit(ProtonSubstringTerm &n) override { visitTerm(n); }
-    virtual void visit(ProtonSuffixTerm &n) override { visitTerm(n); }
-    virtual void visit(ProtonPredicateQuery &) override { }
-    virtual void visit(ProtonRegExpTerm &n) override { visitTerm(n); }
+    void visit(ProtonAnd &) override {}
+    void visit(ProtonAndNot &) override {}
+    void visit(ProtonNear &) override {}
+    void visit(ProtonONear &) override {}
+    void visit(ProtonOr &) override {}
+    void visit(ProtonRank &) override {}
+    void visit(ProtonWeakAnd &) override {}
+
+    void visit(ProtonWeightedSetTerm &n) override { visitTerm(n); }
+    void visit(ProtonDotProduct &n) override { visitTerm(n); }
+    void visit(ProtonWandTerm &n) override { visitTerm(n); }
+    void visit(ProtonPhrase &n) override { visitTerm(n); }
+    void visit(ProtonSameElement &n) override { visitTerm(n); }
```

Comment:
the search iterator is (currently) being modeled without match data, which means that we either need to add match data to the iterator or remove the 'term-ness' from the query node.